src/components/ProgressIndicator/ProgressIndicator.tsx

```diff
@@ -0,0 +1,129 @@
+import React from "react";
+import {
+  Box,
+  CircularProgress as ChakraCircularProgress,
+  CircularProgressLabel as ChakraCircularProgressLabel,
+  Progress as ChakraProgress,
+  useMultiStyleConfig,
+} from "@chakra-ui/react";
+
+import {
+  ProgressIndicatorSizes,
+  ProgressIndicatorTypes,
+} from "./ProgressIndicatorTypes";
+import generateUUID from "../../helpers/generateUUID";
+
+export interface ProgressIndicatorProps {
+  /** Flag to render the component in a dark background. */
+  darkMode?: boolean;
+  /** ID that other components can cross reference for accessibility purposes. */
+  id?: string;
+  /** Whether the `ProgressIndicator` should be linear or circular. */
+  indicatorType?: ProgressIndicatorTypes;
+  /** Whether the progress animation should display because the `value` prop is
+   * not known. When this is set to `true`, the `value` prop will be ignored. */
+  isIndeterminate?: boolean;
+  /** The text to display in an HTML `label` element. */
+  labelText: string;
+  /** Visually show or hide the label text. When set to `false`, then the label
+   * text will be added to the parent component as its `aria-label` attribute. */
+  showLabel?: boolean;
+  /** The size of the linear or circular progress. */
+  size?: ProgressIndicatorSizes;
+  /** A number between 0 to 100 that defines the progress' value. */
+  value?: number;
+}
+
+/**
+ * A component that displays a progress status for any task that takes a long
+ * time to complete or consists of multiple steps. Examples include downloading,
+ * uploading, or processing.
+ */
+const ProgressIndicator: React.FC<ProgressIndicatorProps> = (
+  props: ProgressIndicatorProps
+) => {
+  const {
+    darkMode = false,
+    id = generateUUID(),
+    indicatorType = ProgressIndicatorTypes.Linear,
+    isIndeterminate = false,
+    labelText,
+    showLabel = true,
+    size = ProgressIndicatorSizes.Default,
+    value = 0,
+  } = props;
+  const styles = useMultiStyleConfig("ProgressIndicator", {
+    darkMode,
+    isIndeterminate,
+    size,
+  });
+  let finalValue = value;
+  if (finalValue < 0 || finalValue > 100) {
+    console.warn(
+      "ProgressIndicator: pass in a `value` between 0 and 100. Defaulting to 0."
+    );
+    finalValue = 0;
+  }
+  const progressProps = {
+    // If the label is visually shown, associate it with the progress indicator.
+    // Otherwise, the `aria-label` will be added.
+    ["aria-label"]: showLabel ? null : labelText,
+    ["aria-labelledby"]: showLabel ? `${id}-label` : null,
+    // If `isIndeterminate` is true, then it overrides the `value` prop.
```

Comment:
If a user passes in `isIndeterminate = false` but no value, should they get a warning?

Reply:
Right now, the default value for `isIndeterminate` is `false` and the default for `value` is `0`. If a user passes in `isIndeterminate={false}` and no value, the display will just be the gray bar at value 0. We could hint to the user that maybe they forgot to pass in a `value` prop but I'm not sure if that's necessary.

Reply:
Ah - yes that should be sufficient